### DISCUSSION TITLE
How to map SNOMED-CT and UCUM?

### DISCUSSION THREAD

**Discussion by CDSSAdaptation** [ucum-org/ucum#303] [Q&A]
I want to link/map SNOMED-CT and UCUM together to extract the details of units of measurement. Please let me know the Primary key in both files (SNOMED-CT and UCUM) on which I can do linking.

**Comment by chgessner** [comment]
Unlike SNOMED concepts, measurement units are constructed using a small number of "base units" or "atomic units" using purely mathematical operations. UCUM follows this logic by defining computer-processable codes for those "base units" and "atomic units" together with a syntax that allows for the construction of any "derived unit", and in some cases mathematical "conversion rules" that link certain "derived units", e. g. cm and inch. The latter rules may include numerical factors that are based on conventions external to UCUM. "atomic units" often provide codes for commonly used expressions, i. e. pre-coordinated concepts (example: N for Newton, which is equivalent to the post-coordinated expression kg.m/s2).

Therefore, there is no "primary key" and more importantly no finite list of all possible syntactically valid UCUM expressions. New codes for pre-coordinated expressions and new "atomic units" are added from time to time to UCUM, based on a consensus process described on the pages of ucum.org.

Of course, you can find many of the UCUM "base units" and "atomic units" also in SNOMED CT, e. g. "newton" (SCTID: 278490008). However, SNOMED CT neither captures the mathematical relationship between the various concepts under "Unit of measure" (SCTID: 767524001) nor provides a comparable mechanism for the (mathematical) post-coordination of different unit concepts. This implies that while SNOMED "knows" that the concepts "cm" (SCTID: 258672001) and "mm" (SCTID: 258673006) are both subclasses of "Unit of length" (SCTID: 258667005), SNOMED does not "know" that 1 cm is equivalent to 10 mm. 

Conclusions: 
- You can find corresponding SNOMED concepts for a subset of the "base units" and "atomic units" that provide the definitional basis of the UCUM code system, i. e. the codes defined in https://github.com/ucum-org/ucum/blob/main/ucum-essence.xml . 
- You can find or construct valid UCUM codes for any unit of measure that is also referenced as a SNOMED concept, mostly filed under SCTID: 767524001.
- There is no SNOMED CT relationship type that corresponds to the mathematical relationship between unit concepts, as it is provided implicitly or explicitly by UCUM.
- Unlike SNOMED, UCUM does not provide a classification hierarchy of units (like e. g. the children of SNOMED 767524001 | Unit of measure), besides the mathematical relationship to corresponding base units or to their mathematically derived constructs (e. g. for mass concentration).
- The number of valid UCUM codes is infinite. From a user perspective, UCUM does not make any difference between pre- and post-coordinated expressions. The post-coordination syntax, i. e. the set of construction rules for valid UCUM codes, is considered an integral, inseparable part of UCUM. This is unlike SNOMED, where you have an obvious difference between pre- and post-coordinated SNOMED expressions.

**Reply by CDSSAdaptation** [reply]
Thanks for your reply.

How will it be helpful for converting the units from one measurement unit to another?

For example, in any clinical dataset, if the weight of the patient is mentioned in kg,  that report needs to be transferred to another country for consulting purposes, where the weight is measured in the pound. In such a situation, how can we use UCUM to get the conversion logic or knowledge to convert the kgs to pounds?

**Reply by CDSSAdaptation** [reply]
@chgessner @timbrisc  How can I get the "logic information to convert the unit of measurement" in UCUM?
 For example, in any clinical dataset, if the weight of the patient is mentioned in kg,  that report needs to be transferred to another country for consulting purposes, where the weight is measured in the pound. In such a situation, how can we use UCUM to get the conversion logic or knowledge to convert the kg to pound?

**Reply by chgessner** [reply]
Good question. My answer has two parts: 
1) The formal definition of any unit in UCUM follows the rules laid out in https://ucum.org/ucum . In this case particularly Paragraph 20. The corresponding definitional expressions are listed in the UCUM publication in XML "ucum essence" https://github.com/ucum-org/ucum/blob/main/ucum-essence.xml 
2) As this may be hard to grasp for the occasional user, maybe the best way to make use of the UCUM definition for "pound" is to look up the legal definition of the pound, e. g. in wikipedia. Then come back to the said UCUM specification and trace down the corresponding UCUM definitions from [lb_av] (the code for "pound") to the SI base unit for mass. You will find all required conversion factors in "ucum-essence.xml" (and also in the textual definition https://ucum.org/ucum ). Hint: Look for the word "Avoirdupois", read paragraph 39 of the UCUM specification.
